### ISSUE TITLE
payload loader loads items wrongly

### ISSUE THREAD

**Issue by Hexrotor** [Anuken/Mindustry#6170]
**Platform**: Android

**Build**: 22053

**Issue**: For example the container has 300 capacity of each item.
But if you put a container into payload loader and try to input more than one kind of items, it only loads about 300 items in total so the container won't be full.

**Steps to reproduce**: 
Place it like this and try to count items.

![Screenshot_20211016-214459358_1](https://user-images.githubusercontent.com/45112661/137589920-bbf8d827-e80f-46bf-97a4-886cd9adc39a.jpg)

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [x] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [x] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by Anuken** [comment]
This is not a bug. A container can hold 300 of *any* item, it is "technically" not full until it hold 300 of *every item in the game.* Doing so would be useless behavior.

**Comment by Anuken** [comment]
> He meant that the container only accept 300 items in total.
Which means 100 lead, 100 titanium and 100 graphite, but not 300 each.
But technically, it is not full until it hold 300 of each item.
So this is a bug.

I knew exactly what he meant, and my answer remains the same.

> But technically, it is not full until it hold 300 of each item.

If it actually worked that way, **it would never leave the loader**, because it wouldn't have 300 spore pods, 300 surge alloy, etc.

**Comment by Hexrotor** [comment]
> If it actually worked that way, **it would never leave the loader**, because it wouldn't have 300 spore pods, 300 surge alloy, etc.

Oh fine I understand, so if one day someone finds a way to solve this, I'll be very grateful.😂

**Comment by Hexrotor** [comment]
How about just ignore empty items?
If totolitems = 0, wait for input
If an item is inputted, wait until this item is full and ignore the other empty items. During this if other items are inputted, wait until all they are full, and also ignore the other empty items.
I'm not a programmer so I don't know if this is hard to do, if so then ignore this.

**Comment by Anuken** [comment]
> To be more specific, clicking on the payload loader will display items of the payload in it

That's not going to happen. Those are two different blocks with different inventories.

**Comment by esteemator** [comment]
What if the loading is considered complete if ANY resource is at 300?

**Comment by Hexrotor** [comment]
> How about just ignore empty items? If totolitems = 0, wait for input If an item is inputted, wait until this item is full and ignore the other empty items. **During this if other items are inputted, wait until all they are full**, and also ignore the other empty items. I'm not a programmer so I don't know if this is hard to do, if so then ignore this.

@esteemator

**Comment by Hexrotor** [comment]
> what if you happen to get some sand in the container? What if you only have, say, 100 lead but 300 copper? Any excess resources will freeze everything.

I know that and I just found another problem. If I load water to liquid container first but not full, then load other liquid like cryofluid, it will also get stuck because the liquid container needs water but the loader is full of cryofluid.

![Screenshot_20211017-091946582_1](https://user-images.githubusercontent.com/45112661/137606703-5b076914-1082-4954-a3e0-938ac75a7aea.jpg)


So if you change its working method as I said, maybe this could be solved.
I mean:
>The displayed items are in the loader, but they will be loaded into the payload when the loader is full (item cap depends on the payload in it). And also load them into the payload when a unit is taking it.

So if I inputted some water then input cryofluid, the loader won't accept it. Seems good.
I don't know if this working way will cause some bad things, forgive me pls.

**Comment by Anuken** [comment]
> first but not full, then load other liquid like cryofluid, it will also get stuck because the liquid container needs water but the loader is full of cryofluid.

That's not a bug or even a real problem, no reasonable design mixes liquids like that. Any liquid network would choke if you tried to mix liquids in such a way.

> The displayed items are in the loader, but they will be loaded into the payload when the loader is full (item cap depends on the payload in it). And also load them into the payload when a unit is taking it.

I'm not implementing this.

**Comment by Hexrotor** [comment]
OK I tried the latest version, and I noticed that the loader can only load up to 1200 items in total, regardless of whether there are more than 4 kinds of items(4x300=1200).
Strangely, the last item(plastanium) is the least loaded, and the second last item(silicon) can often be filled. I placed the item source first then placed the loader, so the items are loaded at the same time. 
But anyway, it works better than before :)
![Screenshot_20211017-113956940_1_1](https://user-images.githubusercontent.com/45112661/137610687-dd443716-26fb-467a-a1e3-6c4495100136.jpg)
![Screenshot_20211017-113953534_1_1](https://user-images.githubusercontent.com/45112661/137610696-4f2ec099-f091-4826-b086-7b04b2a83d4e.jpg)
